### PR TITLE
Use Base16 for byte address encoding to String

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
Use `coop.rchain.crypto.codec.Base16` instead of `map("%02x".format(_))`

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/CORE-671

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/node.scala [hunk 1]
```diff
@@ -89,15 +89,14 @@ class NodeRuntime(conf: Conf)(implicit scheduler: Scheduler) {
...
-      }
-      .map("%02x".format(_))
-      .mkString
+    Base16.encode(
+      publicKey
+        .flatMap(CertificateHelper.publicAddress)
+        .getOrElse {
+          println("Certificate must contain a secp256r1 EC Public Key")
+          System.exit(1)
+          Array[Byte]()
```
Comment (goral09) [inline node/src/main/scala/coop/rchain/node/node.scala]: Maybe split that to the logic of extracting public key and encoding it in base16. Also I don't like the `System.exit(1)` in `getOrElse` clause. Would the default empty array from line `98` ever by returned if there's a `System.exit(1)` line before that?

File: comm/src/main/scala/coop/rchain/comm/PeerNode.scala [hunk 2]
```diff
@@ -3,10 +3,12 @@ package coop.rchain.comm
 import java.net.InetSocketAddress
 import com.netaporter.uri.Uri
 import scala.util.control.NonFatal
+import coop.rchain.crypto.codec.Base16
 
 // TODO: Add Show instance
 final case class NodeIdentifier(key: Seq[Byte]) {
-  override def toString: String = key.map("%02x".format(_)).mkString
+  private val keyString         = Base16.encode(key.toArray)
```
Comment (rabbitonweb) [inline comm/src/main/scala/coop/rchain/comm/PeerNode.scala]: this field is used only here, why dont keep it inlined as it was before?

Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/PeerNode.scala]: It's an optimization for performance. We don't want to encode the bytes each time the program calls `toString`.

Comment (rabbitonweb) [reply comm/src/main/scala/coop/rchain/comm/PeerNode.scala]: 👍

**Pull request merged**